src/app/activities/[activityId]/sections/MapSection.js

```diff
@@ -0,0 +1,77 @@
+"use client";
+
+import {useParams} from "next/navigation";
+import {useMutation, useQuery, useQueryClient} from "react-query";
+import React, {useEffect, useState} from "react";
+import {getActivity, patchActivity} from "@/app/activities/[activityId]/api.mjs";
+import Box from "@mui/material/Box";
+import {MapComponent} from "@/app/activities/[activityId]/components/Map";
+import {Button, Container} from "@mui/material";
+import {Errors} from "@/app/activities/[activityId]/components/formikFields";
+
+export function MapSection() {
+  const defaultCoordinates = { lat: 51.5074, lng: -0.1278 };
+  const activityId = useParams().activityId;
+  const queryClient = useQueryClient();
+  const { data: activity } = useQuery(["activity", activityId], () => getActivity(activityId));
+  const [errors, setErrors] = useState([]);
+
+  const [coordinates, setCoordinates] = useState(defaultCoordinates);
+  const [address, setAddress] = useState('');
+
+  const mutation = useMutation((data) => patchActivity(activityId, data));
+
+  useEffect(() => {
+    // Synchronizes the map's coordinates and address with the fetched activity data,
+    // ensuring the map correctly represents the current activity's location.
+    if (activity) {
+      const { latitude, longitude, address: activityAddress } = activity;
+      setCoordinates({ lat: parseFloat(latitude), lng: parseFloat(longitude) });
+      setAddress(activityAddress || '');
+    }
+  }, [activity]);
+
+  async function handleSubmit() {
+    const data = { latitude: coordinates.lat, longitude: coordinates.lng, address };
+    mutation.mutate(data, {
+      onSuccess: () => {
+        console.log("Success")
+        queryClient.invalidateQueries(['activity', activityId]);
```

Comment:
It should just update query data, not actually invalidate
https://tanstack.com/query/v3/docs/framework/react/guides/updates-from-mutation-responses

const useMutateTodo = () => {
  const queryClient = useQueryClient()

  return useMutation(editTodo, {
    // Notice the second argument is the variables object that the `mutate` function receives
    onSuccess: (data, variables) => {
      queryClient.setQueryData(['todo', { id: variables.id }], data)
    },
  })
}